server/src/main/java/io/crate/action/sql/Sessions.java

```diff
@@ -153,6 +159,14 @@ public Session newSession(@Nullable String defaultSchema, Role authenticatedUser
         }
         sessionSettings.statementTimeout(defaultStatementTimeout);
         sessionSettings.memoryLimit(memoryLimit);
+
+        if (authenticatedUser.sessionSettings() != null) {
+            for (Map.Entry<String, Object> entry : authenticatedUser.sessionSettings().entrySet()) {
+                SessionSetting<?> setting = sessionSettingRegistry.settings().get(entry.getKey());
+                setting.apply(sessionSettings, entry.getValue());
```

Comment:
can setting be null?

Reply:
the value or the setting itself? The setting itself no, but Ι will double check.

Reply:
setting itself - but I also think shouldn't be

Reply:
it can't be null because the parser expects a generic property there.